flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/ExecutionContext.java

```diff
@@ -310,6 +316,14 @@ private static Configuration createExecutionConfig(
 		Configuration executionConfig = activeCommandLine.applyCommandLineOptionsToConfiguration(
 				commandLine);
 
+		try {
+			final ProgramOptions programOptions = new ProgramOptions(commandLine);
+			final ExecutionConfigAccessor executionConfigAccessor = ExecutionConfigAccessor.fromProgramOptions(programOptions, dependencies);
+			ConfigUtils.encodeCollectionToConfig(executionConfig, PipelineOptions.JARS, executionConfigAccessor.getJars(), URL::toString);
+		} catch (CliArgsException e) {
+			throw new SqlExecutionException("Invalid deployment run options.", e);
+		}
```

Comment:
You can simplify this into one line:

```java
		ConfigUtils.encodeCollectionToConfig(
			executionConfig,
			PipelineOptions.JARS,
			dependencies,
			URL::toString);
```

We don't need to parse `commandLine` again here.

Reply:
Actually both the `ProgramOptions` and the `executionConfig` are needed for the `flinkConfig` to be complete. This is the same as the `CliFrontend.getEffectiveConfiguration()` method.

So I think that the re-parsing is correct but the:

```ConfigUtils.encodeCollectionToConfig(executionConfig, PipelineOptions.JARS, executionConfigAccessor.getJars(), URL::toString);```

should become:

```executionConfigAccessor.applyToConfiguration(executionConfig);``` (in this order). 

And the `ExecutionConfigAccessor.applyToConfiguration()` should become public.

Reply:
@wuchong @kl0u thanks for your review, I'll update a new version soon.